src/blog/2024/09/node-red-version-control-with-snapshots.md

```diff
@@ -0,0 +1,238 @@
+image: 
+tags:
+   - posts
+   - node red version control
+   - nodered version control
+   - node-red flow backup
+---
+
+Version control is essential, especially when multiple people are working on the same Node-RED project. Without it, changes can easily overlap, or worse—accidental updates could break critical flows. FlowFuse solves this challenge with snapshots, allowing you to create backups of your flows, restore previous versions, and safeguard your project from unexpected issues. 
+
+<!--more-->
+
+In this guide, we’ll show you how to use snapshots in FlowFuse to manage your Node-RED projects with confidence and prevent costly mistakes.
+
+## What is Version Control and Snapshots
+
+Using **version control** for Node-RED flows can introduce complexity and effort, as it often requires frequent pushes of changes to keep everything in sync. This is especially challenging in environments where modifications occur rapidly and continuously.
+
+**Snapshots** simplify this process by providing point-in-time backups of your flows. In the context of Node-RED, snapshots automatically capture the state of your work, ensuring that you can quickly restore them if needed. It captures the following:
+
+- **Flows**: The flow, including all nodes and config nodes of your Node-RED flows.
+- **Credentials**: Any sensitive information used within flows using config nodes.
+- **Environment Variables**: Environment variables you have used or defined within that Node-RED instance.
+- **NPM Packages**: The npm packages you have installed, including custom nodes and Node.js packages.
+- **Runtime Settings**: The configurations that govern the behavior of your Node-RED runtime.
+
+With snapshots in [FlowFuse](/), you can focus on developing your projects while having the confidence that your work is protected.
+
+## Managing Snapshots in FlowFuse
+
+Creating snapshots in FlowFuse is straightforward and can be done in just a few steps.
+
+### Creating Snapshots for Cloud and Device Instances
+
+With this understanding, let’s walk through the steps to create a snapshot in FlowFuse. But before that, let’s understand the differences between a cloud instance and a device instance.
+
+- **Cloud Instance**: This is a Node-RED instance running on a server's virtual machine.
+
+- **Device Instance**: This refers to a Node-RED instance running on your edge devices, which are connected to the FlowFuse platform through the [FlowFuse Device Agent](/product/device-agent/).
+
+Taking snapshots for cloud instances and device instances is completely different. When a device is not assigned to an instance, it will have the same interface and the snapshot tab for creating snapshots, allowing you to create the snapshot in a similar way as you would for an cloud instance. However, once you assign any device to the cloud instance, you will need to follow a different procedure to create device snapshots manually.
+
+#### Creating Snapshots for Cloud Instance
```

Comment:
since we state above that the experience is almost the same for application devices and cloud instances, should this title and the below content not reflect that?

I mention this because there are 2 major sections in this article. 1. "Creating Snapshots for Cloud Instance", 2. "Creating Snapshot for Device Instances when assigned to Cloud Instance".  i.e. there is no _3. Creating Application Device snapshots_